File: exercise/java/week03/exercise2/src/main/java/cache/Cache.java
package cache;

import java.util.HashMap;
import java.util.Map;

public class Cache {
    private Map<String, Integer> map = new HashMap<String, Integer>();

    public int getOrInsert(String key, int value) {
        if (!map.containsKey(key)) {
            map.put(key, value);
        }
        return map.get(key); // Modifies state if key not present and returns value
    }
}

File: exercise/ts/week03-cqs/exercise2/src/cache.ts
export class Cache {
    private map: Map<string, number> = new Map<string, number>();

    getOrInsert(key: string, value: number): number {
        if (!this.map.has(key)) {
            this.map.set(key, value);
        }
        return this.map.get(key)!; // Modifies state if key not present and returns value
    }
}
